dms/models/directory.py

```diff
@@ -494,19 +492,17 @@ def _check_directory_recursion(self):
             raise ValidationError(_("Error! You cannot create recursive directories."))
         return True
 
-    @api.constrains("is_root_directory", "root_storage_id", "parent_id")
+    @api.constrains("is_root_directory", "storage_id", "parent_id")
     def _check_directory_storage(self):
```

Comment:
Shouldn't this checks be in separate constraints? They are testing different things, and maybe we should avoid calling this method too many times...